### ISSUE TITLE
Animation has approx 1 sec delay before starting

### ISSUE THREAD

**Issue by jeghers** [FormidableLabs/victory#848]
When animation is turned on, it seems a long delay of about 1 second occurs before the animation begins (leaving the chart blank and empty for an awkward amount of time, no settings I can find get rid of it.

    <VictoryChart
      responsive={false}
      domain={{ x: [0, 10] }}
      domainPadding={{ y: 80 }}
    >
      <VictoryStack
        horizontal
        colorScale={["lightgreen", "red"]}
        animate={{
          onLoad: { delay: 0, duration: 800 },
          easing: "expInOut",
        }}
        style={{
          data: { width: 45, opacity: 0.7 },
          labels: { fontSize: 14 },
          parent: { border: "1px solid #ccc" }
        }}
      >
        <VictoryBar
          categories={{ x: [ "cats", "dogs" ] }}
          data={[{ x: "cats", y: 6 }, { x: "dogs", y: 3 }]}
        />
        <VictoryBar
          categories={{ x: [ "cats", "dogs" ] }}
          data={[{ x: "cats", y: 3 }, { x: "dogs", y: 5 }]}
        />
      </VictoryStack>
    </VictoryChart>

**Comment by bewildergeist** [comment]
Try setting a `duration` setting on top-level `animate` object too (i.e. sibling to `onLoad` and `easing`). That seems to do the trick for me, oddly enough.

**Comment by s-kris** [comment]
> When animation is turned on, it seems a long delay of about 1 second occurs before the animation begins (leaving the chart blank and empty for an awkward amount of time, no settings I can find get rid of it.
> 
> ```
> <VictoryChart
>   responsive={false}
>   domain={{ x: [0, 10] }}
>   domainPadding={{ y: 80 }}
> >
>   <VictoryStack
>     horizontal
>     colorScale={["lightgreen", "red"]}
>     animate={{
>       onLoad: { delay: 0, duration: 800 },
>       easing: "expInOut",
>     }}
>     style={{
>       data: { width: 45, opacity: 0.7 },
>       labels: { fontSize: 14 },
>       parent: { border: "1px solid #ccc" }
>     }}
>   >
>     <VictoryBar
>       categories={{ x: [ "cats", "dogs" ] }}
>       data={[{ x: "cats", y: 6 }, { x: "dogs", y: 3 }]}
>     />
>     <VictoryBar
>       categories={{ x: [ "cats", "dogs" ] }}
>       data={[{ x: "cats", y: 3 }, { x: "dogs", y: 5 }]}
>     />
>   </VictoryStack>
> </VictoryChart>
> ```

can confirm

**Comment by BradleyJrichardson** [comment]
anyone with a fix?

**Comment by iwan-uschka** [comment]
Any news on this one? The delay is still present in version 34.3.11.

**Comment by gildas-succubus** [comment]
I've fixed that issue like this : 
`<VictoryChart animate={{onLoad: { delay: 0, duration: 1 }, duration: 500}} >`

The duration of the effect is 500ms. Hope this helps

**Comment by iwan-uschka** [comment]
I thought i gave that a try already 🤔 but yes, this does the trick. Thanks @gildas-succubus

**Comment by beccanelson** [comment]
This is something that confused me as well when I was debugging #1997. There seems to be a bit of weirdness around which values Victory uses to determine `onLoad` transitions and delays, and it's worth investigating more.

**Comment by becca-bailey** [comment]
Closing in favor of this larger issue https://github.com/FormidableLabs/victory/issues/2104

**Comment by Dalcor** [comment]
Any update about it?
